src/eventhandler/mod.rs

```diff
@@ -0,0 +1,85 @@
+//! Helper to run a callback in the libuv main thread.
+
+use std::mem;
+use std::os::raw::c_void;
+
+use types::*;
+use handle::{Handle, Managed};
+use context::*;
+use neon_runtime;
+use neon_runtime::raw;
+use std::sync::Arc;
+
+struct EventHandlerInner(*mut c_void);
+
+unsafe impl Send for EventHandlerInner {}
+unsafe impl Sync for EventHandlerInner {}
+
+impl Drop for EventHandlerInner {
+    fn drop(&mut self) {
+        unsafe {
+            neon_runtime::threadsafecb::delete(self.0);
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct EventHandler(Arc<EventHandlerInner>);
+
+impl EventHandler {
+    pub fn new(callback: Handle<JsFunction>) -> Self {
+        TaskContext::with(|mut cx: TaskContext| {
+            let cb = unsafe {
+                neon_runtime::threadsafecb::new(cx.global().to_raw(), callback.to_raw())
+            };
+            EventHandler(Arc::new(EventHandlerInner(cb)))
+        })
+    }
+
+    pub fn bind<T: Value>(this: Handle<T>, callback: Handle<JsFunction>) -> Self {
+        let cb = unsafe {
+            neon_runtime::threadsafecb::new(this.to_raw(), callback.to_raw())
+        };
+        EventHandler(Arc::new(EventHandlerInner(cb)))
+    }
+
+    pub fn schedule<T, F>(&self, arg_cb: F)
+        where T: Value,
+              F: for<'a> FnOnce(&mut TaskContext<'a>) -> Vec<Handle<'a, T>>,
```

Comment:
I'd love to have `IntoIterator<...>` instead of `Vec<..>`, but I couldn't get it to work with the lifetimes....

Reply:
This could be a future improvement. I think to benefit from this, it would require a more holistic change to arguments in Neon.